cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala

```diff
@@ -83,7 +83,11 @@ class AnalyseExecutor(toolInput: Option[String],
 
     formatter.end()
 
-    analysisResult
+    analysisResult.map { result =>
+      val (metricsResults, issuesResults, duplicationResults) =
+        result.partitionSubtypes[MetricsToolExecutorResult, IssuesToolExecutorResult, DuplicationToolExecutorResult]
+      MetricsToolExecutor.reduceMetricsToolResultsByLanguage(metricsResults) ++ issuesResults ++ duplicationResults
```

Comment:
What you are actually doing is merging metrics by file. The language is just a side effect that you need to achieve the first and is not important for the caller. Does this make sense?